rules_ruby_gapic/ruby_gapic.bzl

```diff
@@ -0,0 +1,191 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Rules for generating ruby code with protoc
+including rules for ruby, grpc-ruby, and various flavors of gapic-generator-ruby
+"""
+load("@com_google_api_codegen//rules_gapic:gapic.bzl", "proto_custom_library", "GapicInfo")
+
+##
+# An implementation of the _ruby_gapic_library_add_gapicinfo
+#
+def _ruby_gapic_library_add_gapicinfo_impl(ctx):
+  return [
+    DefaultInfo(files = depset(direct = [ctx.file.output])),
+    GapicInfo(),
+  ]
+
+##
+# A rule that adds the GapicInfo provider to the input
+# used to mark the result of the gapic-generator-ruby plugin
+# to distinguish it from the output of other plugins (e.g. grpc)
+#
+_ruby_gapic_library_add_gapicinfo = rule(
+  _ruby_gapic_library_add_gapicinfo_impl,
+  attrs = {
+    "output": attr.label(allow_single_file = True)
+  }
+)
+
+##
+# A macro over the proto_custom_library that generates a library
+# using the gapic-generator-ads entrypoint
+#
+# name: name of the rule
+# srcs: proto files wrapped in the proto_library rule
+# yml_configs: a list of labels of the yaml configs (or an empty list)
+#
+def ruby_gapic_ads_library(name, srcs, yml_configs, **kwargs):
```

Comment:
Lets discuss the presence of ads-specific rules here over GVC. I believe it is a serious issue.